backend/src/services/activityService.ts

```diff
@@ -420,13 +424,56 @@ export default class ActivityService extends LoggingBase {
         data.username = data.member.username[data.platform][0].username
       }
 
-      const activityExists = await this._activityExists(data, transaction)
+      let activityExists = await this._activityExists(data, transaction)
+
+      const memberService = new MemberService(this.options)
 
-      const existingMember = activityExists
-        ? await new MemberService(this.options).findById(activityExists.memberId, true, false)
+      let existingMember = activityExists
+        ? await memberService.findById(activityExists.memberId, true, false)
         : false
 
-      const member = await new MemberService(this.options).upsert(
+      if (existingMember) {
+        // let's look just in case for an existing member and if they are different we should log them because they will probably fail to insert
+        const tempExisting = await memberService.memberExists(data.member.username, data.platform)
+
+        if (!tempExisting) {
+          logger.warn(
+            {
+              existingMemberId: existingMember.id,
+              username: data.username,
+              platform: data.platform,
+              activityType: data.type,
+            },
+            'We have found an existing member but actually we could not find him by username and platform!',
+          )
+          errorDetails.reason = 'activity_service_createWithMember_existing_member_not_found'
+          errorDetails.details = {
+            existingMemberId: existingMember.id,
+            existingActivityId: activityExists.id,
+            username: data.username,
+            platform: data.platform,
+            activityType: data.type,
+          }
+        } else if (existingMember.id !== tempExisting.id) {
+          logger.warn(
+            {
+              existingMemberId: existingMember.id,
+              actualExistingMemberId: tempExisting.id,
+              existingActivityId: activityExists.id,
+              username: data.username,
+              platform: data.platform,
+              activityType: data.type,
+            },
+            'We found a member with the same username and platform but different id! Deleting the activity and continuing as if the activity did not exist.',
+          )
+
+          await ActivityRepository.destroy(activityExists.id, this.options)
```

Comment:
I think this call is soft-deleting it. Do u think we should hard delete instead?